src/compiler/transformers/es2015.ts

```diff
@@ -1825,6 +1818,10 @@ namespace ts {
             }
 
             if (isBlock(body)) {
+                // ensureUseStrict is false because no new prologue-directive should be added.
+                // addPrologueDirectives will simply put already-existing directives at the beginning of the target statement-array
+                const statementOffset = addPrologueDirectives(statements, body.statements, /*ensureUseStrict*/ false, visitor);

```

Comment:
This also would result in a `"use strict"` appearing after executable statements.